### PR TITLE
RCIdentityAnalysis: some refactoring to improve clarity.

### PR DESCRIPTION
**Pull request by eeckstein**
NFC.

This is a follow-up of https://github.com/apple/swift/pull/34183

### CODE REVIEW

**Comment by eeckstein** [comment]
@swift-ci smoke test

**Comment by eeckstein** [comment]
@swift-ci smoke test linux

File: lib/SILOptimizer/Analysis/RCIdentityAnalysis.cpp [hunk 1]
```diff
@@ -34,6 +34,21 @@ static bool isNoPayloadEnum(SILValue V) {
...
 }
 
+/// Returns true if V is a valid operand of a cast which preserves RC identity.
+///
+/// V is a valid operand if it has a non-trivial type.
+/// RCIdentityAnalysis must not look through casts which cast from a trivial
+/// type, like a metatype, to something which is retainable, like an AnyObject.
+/// On some platforms such casts dynamically allocate a ref-counted box for the
+/// metatype. Naturally that is the place where a new rc-identity begins.
+/// Therefore such a cast is introducing a new rc identical object.
```
Comment (gottesmm) [inline lib/SILOptimizer/Analysis/RCIdentityAnalysis.cpp:44]: Nice comment!

**Review by gottesmm — COMMENTED** [review]
Looks good! Thanks!

**Pull request merged**